backend/routes/download.js

```diff
@@ -34,13 +37,124 @@ const ensureAdminDownloadToken = (req, res, next) => {
   })
 }
 
+router.route('/csv/:showId')
+  .get(ensureAdminDownloadToken, (req, res, next) => {
+    // find the show
+    Show.findById(req.params.showId, { rejectOnEmpty: true })
+      .then(show => {
+        // find all image Entries to this show id
+        Entry.findAll({
+          where: { showId: req.params.showId },
+          attributes: { exclude: ['createdAt', 'updatedAt'] }
+        })
+          .then(entries => {
+            // Get the additonal information on the entries by getting
+            // all the ids and then making the call, to reduce calls to the db
+            const imageIds = []
+            const videoIds = []
+            const otherMediaIds = []
+            entries.forEach((entry) => {
+              if (entry.entryType === IMAGE_ENTRY) {
+                imageIds.push(entry.entryId)
+              } else if (entry.entryType === VIDEO_ENTRY) {
+                videoIds.push(entry.entryId)
+              } else if (entry.entryType === OTHER_ENTRY) {
+                otherMediaIds.push(entry.entryId)
+              }
+            })
+
+            return Promise.all([Image.findAll({ where: { id: { $in: imageIds } } }),
+              Video.findAll({ where: { id: { $in: videoIds } } }), OtherMedia.findAll({ where: { id: { $in: otherMediaIds } } })
```

Comment:
I agree with Tina, throw in a `User.findAll(...)` to get all the users so we can include usernames in the csv. Also, groups, so we can include group participants.